### ISSUE TITLE
Proof regressions of CVC5 versus CVC4 on the SPARKNaCl library

### ISSUE THREAD

**Issue by rod-chapman** [cvc5/cvc5#9052]
I have tried proving the SPARKNaCl cryptographic library using CVC5 1.0.0 (from within a wavefront build of SPARK Pro 23.0w supplied by AdaCore).

The library generates 1588 verification conditions, all of which have been proven using at least one of CVC4, Z3 or Alt-Ergo.

CVC4 alone fails to prove 62 VCs unproven.

CVC5 alone fails to prove 120 VCs, so a serious regression for me.

I have not carried out any further detailed analysis at this stage (I honestly do not have the skill to read the SMT files and/or know what's going on...) I can post the SMT files if you like. The SPARKNaCl sources are on GitHub in https://github.com/rod-chapman/SPARKNaCl

All the best,
 Rod Chapman

**Comment by barrettcw** [comment]
Yes, if you could share the 120 VCs that would be very helpful.  You can post a link here or send them to me directly.

**Comment by rod-chapman** [comment]
I'm not sure how to map from the source file/line number reports from GNATProve to the exact SMT files that correspond to the offending VC. I will see if I can work it out. It would be very good if CVC5 could prove _all_ 1588 - they're all true according to Z3 and Alt-Ergo, so a complete proof using a single prover would be real progress!

**Comment by barrettcw** [comment]
If you prefer, you can also just send us all 1588.  I agree that it would be great to prove all of them.  A little easier said than done...

**Comment by rod-chapman** [comment]
OK.. here come 5 files
gnatprove_cvc4.out is the gnatprove.out file from running proof of the whole library with CVC4 (v1.8) only.
gnatprove_cvc5.out is the gnatprove.out file from running proof with CVC5 (1.0.0) only.
onscreen_cvc5.txt is the on-screen output of GNATProve running with CVC5 only showing 120 unproven VCs.
not_proved_cvc5.out is the gnatprove.out file for CVC5 but only lines that contain "not proved"
gpcvc5fails.tar.gz is the entire gnatprove subdirectory containing the SMT files for the 6 packages that contain the unproven VCs.

**Comment by rod-chapman** [comment]
It's 61 Megabytes, so can't upload here... :-(

**Comment by rod-chapman** [comment]
Here are the four small files as a tarball...
[cvc5issue9052.tar.gz](https://github.com/cvc5/cvc5/files/9364046/cvc5issue9052.tar.gz)

**Comment by rod-chapman** [comment]
Here's a DropBox link to the tarball of the SMT files...  does that work for you?
 https://www.dropbox.com/s/v8xcup0uwu62d8a/gpcvc5fails.tar.gz?dl=0

**Comment by barrettcw** [comment]
Yes that works.  Thanks!
